webapp/channels/src/components/user_settings/general/name_section.tsx

```diff
@@ -0,0 +1,246 @@
+            extraInfo = (
+                <span>
+                    <FormattedMessage
+                        id='user.settings.general.field_handled_externally'
+                        defaultMessage='This field is handled through your login provider. If you want to change it, you need to do so through your login provider.'
+                    />
+                </span>
+            );
+        } else {
+            inputs.push(
+                <div
+                    key='firstNameSetting'
+                    className='form-group'
+                >
+                    <label className='col-sm-5 control-label'>
+                        <FormattedMessage
+                            id='user.settings.general.firstName'
+                            defaultMessage='First Name'
+                        />
+                    </label>
+                    <div className='col-sm-7'>
+                        <input
+                            id='firstName'
+                            autoFocus={true}
+                            className='form-control'
+                            type='text'
+                            onChange={updateFirstName}
+                            maxLength={Constants.MAX_FIRSTNAME_LENGTH}
+                            value={firstName}
+                            onFocus={Utils.moveCursorToEnd}
+                            aria-label={formatMessage({id: 'user.settings.general.firstName', defaultMessage: 'First Name'})}
+                        />
+                    </div>
+                </div>,
+            );
+
+            inputs.push(
+                <div
+                    key='lastNameSetting'
+                    className='form-group'
+                >
+                    <label className='col-sm-5 control-label'>
+                        <FormattedMessage
+                            id='user.settings.general.lastName'
+                            defaultMessage='Last Name'
+                        />
+                    </label>
+                    <div className='col-sm-7'>
+                        <input
+                            id='lastName'
+                            className='form-control'
+                            type='text'
+                            onChange={updateLastName}
+                            maxLength={Constants.MAX_LASTNAME_LENGTH}
+                            value={lastName}
+                            aria-label={formatMessage({id: 'user.settings.general.lastName', defaultMessage: 'Last Name'})}
+                        />
+                    </div>
+                </div>,
+            );
+
+            const notifClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
+                e.preventDefault();
+                updateSection('');
+                updateTab('notifications');
+            };
+
+            const notifLink = (
+                <a
+                    href='#'
+                    onClick={notifClick.bind(this)}
```

Comment:
why do we need to bind here

Reply:
Probably copy paste from the previous implementation without thinking. Will remove.